cmd/gapit/memory.go

```diff
@@ -142,81 +149,81 @@ func (verb *memoryVerb) Run(ctx context.Context, flags flag.FlagSet) error {
 				info := val.SparseImageBlock
-				fmt.Printf("\t\tBlock Offset: (%v, %v)\n",
+				fmt.Fprintf(w, "\t\tBlock Offset: (%v, %v)\n",
 					info.XOffset, info.YOffset)
-				fmt.Printf("\t\tBlock Extent: (%v, %v)\n",
+				fmt.Fprintf(w, "\t\tBlock Extent: (%v, %v)\n",
 					info.Width, info.Height)
-				fmt.Printf("\t\tMip Level:    %v\n", info.MipLevel)
-				fmt.Printf("\t\tArray Layer:  %v\n", info.ArrayLayer)
-				if asp := stringifyAspects(info.Aspects); len(asp) > 0 {
-					fmt.Printf("\t\tAspects:      %v\n", asp)
-				}
+				fmt.Fprintf(w, "\t\tMip Level:    %v\n", info.MipLevel)
+				fmt.Fprintf(w, "\t\tArray Layer:  %v\n", info.ArrayLayer)
+				fmt.Fprintf(w, "\t\tAspects:      %v\n", aspectList(info.Aspects))
 			case *api.MemoryBinding_SparseImageMetadata:
 				info := val.SparseImageMetadata
-				fmt.Printf("\t\tArray Layer:     %v\n", info.ArrayLayer)
-				fmt.Printf("\t\tMip Tail Offset: %v\n", info.Offset)
+				fmt.Fprintf(w, "\t\tArray Layer:     %v\n", info.ArrayLayer)
+				fmt.Fprintf(w, "\t\tMip Tail Offset: %v\n", info.Offset)
 			case *api.MemoryBinding_SparseImageMipTail:
 				info := val.SparseImageMipTail
-				fmt.Printf("\t\tArray Layer:     %v\n", info.ArrayLayer)
-				fmt.Printf("\t\tMip Tail Offset: %v\n", info.Offset)
-				if asp := stringifyAspects(info.Aspects); len(asp) > 0 {
-					fmt.Printf("\t\tAspects:      %v\n", asp)
-				}
+				fmt.Fprintf(w, "\t\tArray Layer:     %v\n", info.ArrayLayer)
+				fmt.Fprintf(w, "\t\tMip Tail Offset: %v\n", info.Offset)
+				fmt.Fprintf(w, "\t\tAspects:         %v\n", aspectList(info.Aspects))
 			case *api.MemoryBinding_SparseOpaqueImageBlock:
-				fmt.Printf("\t\tImage Memory Offset: %v\n",
+				fmt.Fprintf(w, "\t\tImage Memory Offset: %v\n",
 					val.SparseOpaqueImageBlock.Offset)
 			case *api.MemoryBinding_SparseBufferBlock:
-				fmt.Printf("\t\tBuffer Memory Offset: %v\n",
+				fmt.Fprintf(w, "\t\tBuffer Memory Offset: %v\n",
 					val.SparseBufferBlock.Offset)
 			}
 		}
 
 		aliases := bindings.computeAliasing()
 		if len(aliases) == 0 {
-			fmt.Println("\tNo aliased regions")
+			fmt.Fprintln(w, "\tNo aliased regions")
 		} else {
-			fmt.Printf("\t%v aliased regions:\n", len(aliases))
+			fmt.Fprintf(w, "\t%v aliased regions:\n", len(aliases))
 			for i, a := range aliases {
-				fmt.Printf("\t%v:\n", i)
-				fmt.Println("\t\tOffset: ", a.offset)
-				fmt.Println("\t\tSize:   ", a.size)
-				fmt.Println("\t\tShared by:")
+				fmt.Fprintf(w, "\t%v:\n", i)
+				fmt.Fprintln(w, "\t\tOffset: ", a.offset)
+				fmt.Fprintln(w, "\t\tSize:   ", a.size)
+				fmt.Fprintln(w, "\t\tShared by:")
 				for _, s := range a.sharers {
-					fmt.Printf("\t\t\t%v\n", s)
+					fmt.Fprintf(w, "\t\t\t%v\n", s)
 				}
 			}
 		}
 	}
+	w.Flush()
 	return nil
 }
 
+type aspectList []api.AspectType
+
+func (a aspectList) Format(f fmt.State, c rune) {
+	if len(a) == 0 {
+		return
+	}
+	strs := make([]string, len(a))
+	for i, asp := range a {
+		var typ string
+		switch asp {
```

Comment:
This can be a format method on `api.AspectType`. I suspect then you don't need the `aspectList` type at all.

Reply:
Ok, and then just print out the slice?

Reply:
Yeah, you'd have to check it formats it like you want though

Reply:
Right, yeah I think I'd want to drop the `[]` since the usual case is just one aspect